NExtends/Collections/DictionaryListMapper.cs

```diff
@@ -0,0 +1,234 @@
+		{
+			this.MapEntries = mapEntries;
+			this.MapEntriesKeys = this.MapEntries.Select(me => me.Key).ToList();
+			this.MapEntriesIndexes = this.MapEntries.Select(me => me.Index).ToList();
+
+			Map = new Dictionary<string, IDictionaryListMapEntry>();
+			foreach (var entry in mapEntries)
+				Map.Add(entry.Key, entry);
+		}
+
+		public void Add(string key, V value)
+		{
+			throw new NotImplementedException();
+		}
+
+		public bool ContainsKey(string key)
+		{
+			return MapEntriesKeys.Contains(key);
+		}
+
+		public ICollection<string> Keys
+		{
+			get { return MapEntriesKeys; }
+		}
+
+		public bool Remove(string key)
+		{
+			throw new NotImplementedException();
+		}
+
+		public bool TryGetValue(string key, out V value)
+		{
+			bool succes = false;
+
+			value = default(V);
+			try
+			{
+				int idx = Map[key].Index;
+				value = CollectionToMap[idx];
+
+				succes = true;
+			}
+			catch
+			{
+
+			}
+			return succes;
+		}
+
+		public ICollection<V> Values
+		{
+			get
+			{
+				List<V> values = new List<V>();
+				foreach (var idx in MapEntriesIndexes)
+					values.Add(CollectionToMap[idx]);
+				return values;
+			}
+		}
+
+		public V this[string key]
+		{
+			get
+			{
+				int idx = Map[key].Index;
+				return CollectionToMap[idx];
+			}
+			set
+			{
+				int idx = Map[key].Index;
+				CollectionToMap[idx] = value;
+			}
+		}
+
+		public void Add(KeyValuePair<string, V> item)
+		{
+			throw new NotImplementedException();
+		}
+
+		public void Clear()
+		{
+			throw new NotImplementedException();
+		}
+
+		public bool Contains(KeyValuePair<string, V> item)
+		{
+			if (!ContainsKey(item.Key))
+				return false;
+
+			return this[item.Key].Equals(item.Value);
+		}
+
+		public void CopyTo(KeyValuePair<string, V>[] array, int arrayIndex)
+		{
+			foreach (var mapEntry in MapEntries)
+			{
+				var kvp = new KeyValuePair<string, V>(mapEntry.Key, this[mapEntry.Key]);
+				array.SetValue(kvp, arrayIndex++);
+			}
+		}
+
+		public int Count
+		{
+			get { return MapEntries.Count(); }
+		}
+
+		public bool IsReadOnly
+		{
+			get { return false; }
+		}
+
+		public bool Remove(KeyValuePair<string, V> item)
+		{
+			throw new NotImplementedException();
+		}
+
+		public IEnumerator<KeyValuePair<string, V>> GetEnumerator()
+		{
+			return new ListDictionaryMapperEnumerator(this);
+		}
+
+		System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator()
+		{
+			return GetEnumerator();
+		}
+
+		public class ListDictionaryMapperEnumerator : IEnumerator<KeyValuePair<string, V>>
+		{
+			DictionaryListMapper<V> Mapper;
+			IEnumerator<IDictionaryListMapEntry> It;
+
+			public ListDictionaryMapperEnumerator(DictionaryListMapper<V> mapper)
+			{
+				this.Mapper = mapper;
+				Reset();
+			}
+
+			public KeyValuePair<string, V> Current
+			{
+				get
+				{
+					var mapEntry = It.Current;
+					return new KeyValuePair<string, V>(mapEntry.Key, Mapper[mapEntry.Key]);
+				}
+			}
+
+			public void Dispose()
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=csharpsquid%3AS1186)